### PR TITLE
Replace read-package-tree with npm's Arborist (close #64)

### LINKED ISSUE #64: Incoming: Arborist
Isaac was kind enough to confirm that `read-package-tree` will probably be deprecated in favor of [Arborist](https://github.com/npm/arborist).  Arborist will expose a promise API.

I have watched release on the Arborist repo.  When the time comes, `licensee` will want to switch. But I'd strongly recommend that we hold off doing any substantial development until that decision is taken.

### PR DESCRIPTION
**Pull request by kemitchell**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: ['10', '11', '12', '13', '14']
+        node: ['12', '13', '14']
```
Comment (ljharb) [inline .github/workflows/ci.yml:8]: this reflects a breaking change that we'll need to make sure is captured in `engines.node` as well

File: index.js [hunk 2]
```diff
@@ -35,29 +33,18 @@ function licensee (configuration, path, callback) {
...
-            }
-            callback(null, findIssues(configuration, children, []))
-          })
-      }
-    )
+    var arborist = new Arborist({ path })
+    arborist.loadActual({ forceActual: true })
+      .catch(function (error) {
+        return callback(error)
+      })
```
Comment (ljharb) [inline index.js:40]: if this doesn't rethrow, it will trigger the next `.then` - so you may need to move it until after the then

Comment (ljharb) [reply index.js]: this doesn't seem resolved yet

File: index.js [hunk 3]
```diff
@@ -37,113 +33,21 @@ function licensee (configuration, path, callback) {
...
-        if (error) callback(error)
-        else {
-          if (configuration.filterPackages) {
-            packages = configuration.filterPackages(packages)
-          }
-          withTrees(packages, false)
+    var arborist = new Arborist({ path })
+    arborist.loadActual({ forceActual: true })
+      .then(function (tree) {
+        var children = Array.from(tree.children.values())
```
Comment (isaacs) [inline index.js:39]: Note, this is only the packages that happen to be installed in the top-level `node_modules` folder.  (Parent/children = tree = physical folder structure.  Edges = dependencies = logical dependency graph.)

If you intend to get a flat list of _every_ package in the tree, regardless of folder hierarchy or dependency graph (including packages that are unreachable or extraneous), you can instead use `var children = Array.from(tree.inventory.values())`

File: index.js [hunk 4]
```diff
@@ -169,62 +73,47 @@ function isString (argument) {
...
-    has(dependencies, key) &&
-    dependencies[key].indexOf(version) !== -1
-  )
-}
-
 function resultForPackage (configuration, tree) {
   var packageAllowlist = configuration.packages || {}
   var result = {
-    name: tree.package.name,
+    name: tree.name,
```
Comment (isaacs) [inline index.js:108]: Subtle edge case here (possibly affecting the use of `tree.name` in the children selection logic above, as well): `tree.name` and `tree.package.name` are not _always_ the same.  `name` is the name by which it is referenced for `import` or `require`, the folder name in `node_modules`.  `tree.package.name` is the name as it appears in `package.json`.

For example, if you do `npm install lodash@npm:underscore@latest`, then `node.name` is `"lodash"`, but `node.package.name` is `"underscore"`.

**Pull request merged**